test/ex_webrtc/peer_connection_test.exs

```diff
@@ -544,22 +581,46 @@ defmodule ExWebRTC.PeerConnectionTest do
     end
   end
 
-  test "can exchange data on single transceiver" do
-    # setup track pc1 -> pc2
-    {:ok, pc1} = PeerConnection.start_link()
-    {:ok, pc2} = PeerConnection.start_link()
+  describe "send data in both directions on a single transceiver" do
+    test "using one negotiation" do
+      {:ok, pc1} = PeerConnection.start_link()
+      {:ok, pc2} = PeerConnection.start_link()
 
-    track1 = MediaStreamTrack.new(:audio)
-    {:ok, _sender} = PeerConnection.add_track(pc1, track1)
+      track1 = MediaStreamTrack.new(:audio)
+      track2 = MediaStreamTrack.new(:audio)
 
-    :ok = negotiate(pc1, pc2)
+      {:ok, _sender} = PeerConnection.add_track(pc1, track1)
+      {:ok, offer} = PeerConnection.create_offer(pc1)
+      :ok = PeerConnection.set_local_description(pc1, offer)
+      :ok = PeerConnection.set_remote_description(pc2, offer)
 
-    # setup track pc2 -> pc1
-    track2 = MediaStreamTrack.new(:audio)
-    {:ok, _sender} = PeerConnection.add_track(pc2, track2)
+      [tr] = PeerConnection.get_transceivers(pc2)
+      :ok = PeerConnection.set_transceiver_direction(pc2, tr.id, :sendrecv)
+      :ok = PeerConnection.replace_track(pc2, tr.sender.id, track2)
 
-    :ok = negotiate(pc2, pc1)
+      {:ok, answer} = PeerConnection.create_answer(pc2)
+      :ok = PeerConnection.set_local_description(pc2, answer)
+      :ok = PeerConnection.set_remote_description(pc1, answer)
+    end
```

Comment:
This does not look like it tests "send data in both (...) using one negotiation" to me.

Reply:
Fixed